packages/fx-core/src/plugins/solution/fx-solution/solution.ts

```diff
@@ -1641,9 +1641,9 @@ export class TeamsAppSolution implements Solution {
     const notifications: string[] = [];
     const pluginsToScaffold: LoadedPlugin[] = [this.LocalDebugPlugin];
     const azureResource = Array.from(settings.azureResources || []);
-    if (addFunc || ((addSQL || addApim) && !alreadyHaveFunction)) {
-      pluginsToScaffold.push(functionPlugin);
+    if (addFunc || ((addSQL || addApim || addKeyVault) && !alreadyHaveFunction)) {
       if (!azureResource.includes(AzureResourceFunction.id)) {
```

Comment:
one question, it uses id to identify the scaffold. If the add is before provision, what will happen?

Reply:
these id read from azureResouruces in projectSettings.json file, provision has no effect on this value.

Reply:
Got it!